graph/src/env/store.rs

```diff
@@ -160,4 +180,42 @@ pub struct InnerStore {
     write_queue_size: usize,
     #[envconfig(from = "GRAPH_STORE_BATCH_TARGET_DURATION", default = "180")]
     batch_target_duration_in_secs: u64,
+    #[envconfig(from = "GRAPH_STORE_HISTORY_COPY_THRESHOLD", default = "0.5")]
+    copy_threshold: ZeroToOneF64,
+    #[envconfig(from = "GRAPH_STORE_HISTORY_COPY_THRESHOLD", default = "0.05")]
+    delete_threshold: ZeroToOneF64,
+    #[envconfig(from = "GRAPH_STORE_HISTORY_SLACK_FACTOR", default = "1.2")]
+    history_slack_factor: HistorySlackF64,
+}
+
+#[derive(Clone, Copy, Debug)]
+struct ZeroToOneF64(f64);
+
+impl FromStr for ZeroToOneF64 {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let f = s.parse::<f64>()?;
+        if f < 0.0 || f > 1.0 {
+            bail!("invalid value: {s} must be between 0 and 1");
+        } else {
+            Ok(ZeroToOneF64(f))
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+struct HistorySlackF64(f64);
+
+impl FromStr for HistorySlackF64 {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let f = s.parse::<f64>()?;
+        if f < 1.01 {
+            bail!("invalid value: {s} must be bigger than 1.01");
```

Comment:
Why can't the slack factor can't be set to 1? If that can cause numerical instability somewhere, imo we should fix that there and allow this to be set to 1.